src/edu/washington/escience/myria/operator/agg/StringAggregator.java

```diff
@@ -1,141 +1,108 @@
 package edu.washington.escience.myria.operator.agg;
 
-import java.util.Objects;
-import java.util.Set;
+import java.util.List;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
-import com.google.common.math.LongMath;
 
 import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
 import edu.washington.escience.myria.storage.AppendableTable;
-import edu.washington.escience.myria.storage.ReadableTable;
+import edu.washington.escience.myria.storage.MutableTupleBuffer;
+import edu.washington.escience.myria.storage.ReadableColumn;
+import edu.washington.escience.myria.storage.ReplaceableColumn;
+import edu.washington.escience.myria.storage.TupleBatch;
 
 /**
  * Knows how to compute some aggregate over a StringColumn.
  */
 public final class StringAggregator extends PrimitiveAggregator {
 
-  /** Required for Java serialization. */
-  private static final long serialVersionUID = 1L;
-  /** Which column of the input this aggregator operates over. */
-  private final int fromColumn;
-
-  /**
-   * Aggregate operations applicable for string columns.
-   */
-  public static final Set<AggregationOp> AVAILABLE_AGG =
-      ImmutableSet.of(AggregationOp.COUNT, AggregationOp.MAX, AggregationOp.MIN);
-
-  /**
-   * @param aFieldName aggregate field name for use in output schema.
-   * @param aggOps the aggregate operation to simultaneously compute.
-   * @param column the column being aggregated over.
-   */
-  public StringAggregator(final String aFieldName, final AggregationOp[] aggOps, final int column) {
-    super(aFieldName, aggOps);
-    fromColumn = column;
+  protected StringAggregator(
+      final String inputName, final int column, final AggregationOp aggOp, final int[] stateCols) {
```

Comment:
Because an Aggregator only performs one op. The factory will generates multiple aggregators if we have multiple ops on one column.